Agregar Código Básico

IIKUYY opened the pull request.

IIKUYY committed:
Agregar Código Básico

semanticmx commented:
Para todos los repos, desde el inicio hay que agregar el archivo del PR template

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,537 @@
+  <head>
+    <title>Template</title>
+    <meta 
+        charset="UTF-8" />
```
cuando haya un solo atributo el atributo va en la misma línea.

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,537 @@
+  <body>
+    <header>
+      <section 
+        id="contacto">
```
hay que evitar el uso de ID, en su lugar podemos usar CLASS o DATA-*

semanticmx commented:
Si accedes a https://semantic.mx y desactivas los estilos de la página vas a ver algo como esto:
<img width="1405" alt="Screen Shot 2023-08-23 at 11 16 07" src="https://github.com/IIKUYY/CSS/assets/2141584/bfcc00df-8c0f-4468-8f8f-1a48cc999623">

La idea de una implementación semántica de HTML es que puedas navegar el sitio aunque no haya ningún estilo, por ejemplo:

1. En lugar de una barra de navegación se ve un listado de links a los que les puedes dar click
2. El nombre del sitio se destaca como headings de diferentes tamaños
3. Cada sección tiene su heading, textos y formatos de contenido

La versión de Almex es similar, solo que no veo el nombre de Almex, en la versión de Semantic Weapons con estilos se ve una imagen y sin estilos se ve el nombre del sitio.

En general títulos y contenido de las secciones se tienen que implementar con las etiquetas HTML que corresponda.

Como regla general el uso de <div> <span> debe ser limitado al igual que anidar más de 3 niveles las etiquetas.

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,537 @@
+    <header>
+      <section 
+        id="contacto">
+        <div 
```
Estos divs hay que reemplazarlos por etiquetas semánticas.

En lugar de:

```html
<div 
            id="contactoSucursal">
          <a 
            href="">
            ubica tu sucursal más cercana
          </a>
        </div>
        <div 
            id="formasContacto">
          <a 
            href="">
            comunícate con nosotros
          </a>
        </div>
```

Se puede usar:

```html
<body>
    <header>
      <section> 
        <nav>
          <li>
            <a>Menu Item</a>
          </li>
          <li>
            <a>Menu Item</a>
          </li>
        </nav>
      <section> 
    <header>
```

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,537 @@
+          <label 
+            for="language">
+            <img
+              src="C:\Users\elier\OneDrive\Documentos\GitHub\CSS\template\image 2.png"
```
Este src no se va a cargar desde internet, implica que tengo que tener Windows, entrar a mi unidad C: y tener una carpeta con esa exacta ruta.

Lo que se tiene que hacer es usar [rutas relativas](https://www.w3schools.com/html/html_filepaths.asp).

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,537 @@
+                <button>
+                    Contacto
+                </button>
+                <div
```
hay que revisar el uso de divs anidados, un div semanticamente no significa nada y la mayoría de las veces se puede reemplazar por una etiqueta que le de más significado al contenido:

```html
<nav>
      <ul>
          <li>
            <label>Envios</label>
            <ul>
              <li>Opcion 1</li>
              <li>Opcion 2</li>
            </ul>
          </li>
          <li>
            <label>Soluciones</label>
            <ul>
              <li>Opcion 1</li>
              <li>Opcion 2</li>
            </ul>
          </li>
      </ul>
```

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,537 @@
+        <div
+            class="carretera"
+            id="bienvenida"
+            style="background-image: url(C:\Users\elier\OneDrive\Documentos\GitHub\CSS\template\S_r6t7Ew.png);">
```
El CSS se puede utilizar desde una hoja de estilo a parte, desde el head con la etiqueta <style> o como atributo con el atributo style.

Lo que queremos es mantener independientes la estructura, el estilo y la funcionalidad.

Entonces no debe haber nada de estilos en el HTML, todos se cargan de una hoja de estilos a parte.

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,537 @@
+        <h5>
+            Empresa familiar
+        </h5>
+        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
```
Igual no debe haber texto suelto, fuera de una etiqueta, en este caso se puede usar la etiqueta <p>

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,537 @@
+                alt="logoBW">
+            Lideres en el servicio de carga desde 1951
+            <a 
+                href="">
```
El href nunca se deja vacío, si no tenemos links a donde mandar un hipervínculo usamos el hash (#)

```html
<a href="#">
  Link
</a>
```

semanticmx reviewed (changes requested):
hay que aplicar los comentarios a todo el código que aplique por favor.

IIKUYY committed:
Chores: Realizar cambios

# what

- Cambiar divs por etiquetas semanticas
- Eliminar texto fuera de etiquetas
- Agregar # a href vacios
- Cambiar imagenes a rutas realtivas
- Cambiar imagenes a carpeta assets
- Mejorar anidación

# why

- Aplicar de manera correcta los consejos dados para tener un código legible y más correcto

# reference

-

IIKUYY commented:
Cambios realizados

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+          content="width=device-width, initial-scale=1" />
+  </head>
+  <body>
+    <header> <! Encabezado
```
No soy muy fan de los comentarios en HTML, si te das cuenta con las etiquetas semánticas ahora el código se podría ver así:

```html
<header> <!-- Encabezado -->
```

o lo que es lo mismo:

```html
<header> <!-- Header -->
```

Es decir, la etiqueta semántica ya ayuda a que evitemos esos comentarios. Antes tenían más sentido porque el código que veía algo así:

```html
<div> <!-- Contenedor -->
  <div> <!-- Encabezado -->
    <div> <!-- Navegación -->
      <div> <!-- Menu Item -->
```

semanticmx replied:
@IIKUYY este falto de corregir:

dice:

```html
<! Encabezado
```

Debería decir:

```html
<!-- Encabezado
```

O mejor eliminar el comentario porque es redundante aquí

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+            </nav>
+        </section>
+    </header>
+    <section> <! Sección de bienvenida
```
El formato para los comentarios es:

```html
<!--
```

Faltan los dos guiones medios.

En este caso puedes dejar el comentario ya que aunque es repetitivo `section` y `sección...` el hecho de que sea la sección de bienvenida le da utilidad así que puede quedar como `<!-- Bienvenida -->`

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+            </button>
+        </article>
+        <article>
+            <iframe src="#" frameborder="3"></iframe>
```
Los iframes generalmente los utilizamos para embedir código que ya existe en otro sitio y que se despliegue en la página "padre".

En el caso del diseño simplemente es un tipo "[card](https://m3.material.io/components/cards/overview)" que muestra el resumen a dos posts recientes, a los cuales al darle click te lleva al blog del sitio.

Estos pueden quedar como `<section>`s del `<article>`

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+                        Lideres en servicios integrales de carga consolidados desde 1951
+                    </P>
+                </li>
+                <section>
```
Este por otro lado en lugar de section puede ser un `<nav>`

Ya que es una lista de links que al darle click te llevan a página externa

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+                    <img src="template\Assets\ALMEX_B&W.png" alt="logoBW">
+                </li>
+                <li>
+                    <P>
```
todas las etiquetas hay que colocarlas en minúsculas

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+                        </a>
+                    </li>
+                </section>
+                <section>
```
Todos estos sections serían 4 navs

IIKUYY committed:
Chores: Realizar cambios

# what

- Realizar los cambios que se requirieron

# why

- Aplicar de manera correcta los consejos dados para tener un código legible y más correcto

# reference

-

IIKUYY commented:
Cambios realizados

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+    <title>Almex Ejemplo</title>
+    <meta charset="UTF-8" />
+    <meta name="viewport" 
+          content="width=device-width, initial-scale=1" />
```
en HTML5 ya le podemos quitar el slash (/) final a las etiquetas vacías

en lugar de:

```
<meta name="viewport" 
           content="width=device-width, initial-scale=1" />
```

queda como:

```
<meta name="viewport" 
           content="width=device-width, initial-scale=1">
```

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+        </address>
+        <section>
+            <figure>
+                <img src="template\Assets\USF.png" 
```
En HTML los separadores de directorios son con slash (/)

dice:

```
<img src="template\Assets\USF.png"
```

debe decir:

```
<img src="template/Assets/USF.png"
```

IIKUYY committed:
Chores: Realizar cambios

# what

- Realizar los cambios que se requirieron

# why

- Aplicar de manera correcta los consejos dados para tener un código legible y más correcto

# reference

-

IIKUYY commented:
Cambios realizados

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+               </button>
+            </address>
+    </section>
+    <section> <! Información general
```
te faltó este de aquí

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+            </section>
+        </aside>
+    </section>
+    <section> <! Galería
```
y este

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+                </svg>
+            </button>
+            <figure>
+                <img src="template\Assets\Carretera_gris.png" 
```
Y etsas rutas también hay que cambiarle la diagonal invertida.

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+                     fill="none" 
+                     xmlns="http://www.w3.org/2000/svg">
+                    <g id="RoundButton_NavPrev">
+                        <g id="Ellipse" 
```
hay que hacer un poco de edición a los ID's al copiar los SVG porque como se pasan con el mismo valor eso no se puede tener en HTML.

Generalmente se pueden eliminar solamente, pero también se pueden editar a algo como `id="Ellipse-Aside-1"` o algo así

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+        </section>
+    </header>
+    <section> <!--Bienvenida-->
+        <main>
```
Solo debe existir una etiqueta <main> en la página y generalmente se usa para webapps, en otros casos rara vez la usamos ya que es como un duplicado del <body>.

Generalmente la estructura es:

```html
<body>
 <header>
  <nav></nav>
 </header>
 <aside></aside>
 <section>
  <article>
   <header...
   <footer...
  </article>
 ...
 <article>
   <header...
   <footer...
  </article>
 </section>
 <nav></nav>
</body>
```

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+            </button>
+        </article>
+        <article>
+            <section src="#" frameborder="3"></section>
```
Los section no llevan el atributo `src`

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+        <article>
+            <section src="#" frameborder="3"></section>
+        </article>
+        <article>
```
Los `article` deben de llevar siempre un heading.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,345 @@
+                        Lideres en servicios integrales de carga consolidados desde 1951
+                    </p>
+                </li>
+                <nav>
```
El `ul` puede ir dentro de un `nav`pero el `nav`no puede ir dentro de un `ul`.

Normalmente la estructura de los footers actuales es usando el tag `footer` como contenedor de la sección.

Luego varias secciones nav como en el caso header+nav, donde cada nav muestra sus links de navegación y son independientes.

Esto aplica para la parte de mobile/responsive ya que si tu tienes un solo bloque (`<ul>...</ul>`) es más difícil partirlo en columnas a que si desde el principio ya tienes muchos bloques (`<nav></nav><nav></nav><nav></nav>`) que puedes sin problema re-acomodar en columnas.

semanticmx reviewed (changes requested):
Puedes revisar este artículo como guía para el armado del layout

https://www.smashingmagazine.com/2009/08/designing-a-html-5-layout-from-scratch/

IIKUYY committed:
Chores: Realizar cambios

# what

- Realizar los cambios que se requirieron

# why

- Aplicar de manera correcta los consejos dados para tener un código legible y más correcto

# reference

-

IIKUYY commented:
Cambios realizados

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,341 @@
+        </section>
+    </header>
+    <section> <!--Bienvenida-->
+        <p>
```
El uso de esta `<p>` esta raro, a veces se utiliza para agregar una línea de separación pero sería así:

```html
<p>&nbsp;</p>
```

`<p>` es una etiqueta contenedora de texto y dentro del texto podemos agregar `<strong>`, `<small>`, `<i>`, etc., pero no headings.

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,341 @@
+                    Comunícate con nosotros
+            </a>
+        </address>
+        <section>
```
Tienes que

a) Buscar todos los `section` y `article` y verificar que usas un heading
o
b) Checar tu layout para ver si necesitas de un section o de otro tag

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,341 @@
+        <h1>
+            Somos lideres
+        </h1>
+        <main>
```
Los tags `<main>` yo no los usaría en esta implementación, si se usan no pueden estar dentro de un section, article o algo así, deben ser los contenedores principales

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,341 @@
+            Desde 1951 hemos sido lore ipsum dolor sit amet .................
+        </main>
+        <article>
+            <h1>
```
Independientemente del tamaño de las fonts en el diseño puedes usar diferentes tipos de headings, si ya usaste un h1 en secciones arriba, las siguientes secciones pueden usar h2

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,341 @@
+                                    fill="white"/>
+                        </g>
+                        <g id="ios-arrow-forward">
+                            <path id="Vector" 
```
Este ID se repite, puedes hacer una búsqueda por `id="` y checar si necesitas ajustar otros

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,341 @@
+    </section>
+    <footer>
+        <nav>
+            <li>
```
acá falta el `ul`

debe quedar así:

```
<footer>
 <nav>
  <ul>
   <li>
    <a>
...
```

semanticmx commented:
@IIKUYY puedes usar validadores, ya sea en tu editor de código o de internet:

https://validator.w3.org

<img width="1075" alt="Screen Shot 2023-08-30 at 12 31 05" src="https://github.com/IIKUYY/CSS/assets/2141584/03124e6d-cff9-4de1-b3f4-b034f88f7a99">

IIKUYY committed:
Chores: Realizar cambios

# what

- Realizar los cambios que se requirieron

# why

- Aplicar de manera correcta los consejos dados para tener un código legible y más correcto

# reference

-

IIKUYY commented:
Cambios realizados

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,338 @@
+  </head>
+  <body>
+    <header>
+        <address>
```
Address se utiliza para direcciones tipo calle,número, etc.

No se usa para los links.

Aquí para reducir complejidad puedes optar por:

```html
<header>
 <a>
  Ubica...
 </a>
 ...
</header>
```

o recuerda que para los links siempre usamos <nav> cuando son dos o más, ya que se considera ya navegación del sitio.

```html
<header>
 <nav><!-- links --></nav>
 <section><!-- idioma --></section>
</header>
```

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,338 @@
+            </select>
+        </section>
+        <section>
+            <h2>
```
igual acá, podemos reducir los contenedores quitando el .h2 y dejando el .img como child del .section

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,338 @@
+                    </li>
+                </ul>
+            </nav>
+        </section>
```
faltó el link de Mi Almex y el icono de búsqueda

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,338 @@
+            <h2>
+                Nuestro Camino es historia
+            </h2>
+            <address>
```
igual, quitamos el .address y solo queda el .button como child del .section

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,338 @@
+               </button>
+            </address>
+    </section>
+    <section>
```
toda esta sección quedó muy bien.

solo faltaría ver cómo se agregan los iconos del check de employees  y trailers. pero eso puede quedar en otro PR ya que llegues a aplicar estilo a esa parte.

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,338 @@
+                Trailers en el camino
+            </p>
+        </article>
+        <aside>
```
aside es más como para barras de navegación laterales.

este puede ser un tercer .article

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,338 @@
+                    Empresa familiar
+                </h2>
+                <p>
+                    Lore ipsum ..........................
```
aquí hay unas palabras en negritas, esas si las podemos colocar para usar un .strong

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,338 @@
+            </section>
+        </aside>
+    </section>
+    <section>
```
Yo estructuraría esto de la siguiente forma:

```html
...
<section class="featured"></section>
<section></section> <!-- principal -->
<section class="noticias"></section>
<footer></footer>
```

como lo tienes está así:

```html
...
<section> <!--Bienvenida--></section>
<section></section> <!-- Somos líderes -->
<section></section> <!-- 1951 -->
<section class="noticias"></section>
<footer></footer>
```

yo agruparía somos líderes y lo de 1951 en una sola sección porque si te das cuenta en el diseño el fondo de todo eso es blanco y cambia a gris hasta noticias.

Ahora como visualmente hay dos columnas y luego una, tus .section quedaría así:

```html
...
<section class="featured"></section>
<div class="container"> <!-- principal -->
 <section class="2-cols"> <!-- agrupas las primeras dos columnas -->
  <h2>Somos líderes</h2>
  <p>Since...</p>
  <article>
   </h3>385</h3>
   <p>Office...</p>
  </article>
  <article>
   </h2>01</h2>
   <h3>Empresa...</h3>
   <p>Cepteur...</p>
   ...
  </article>
 </section>
 <section> <!-- slider -->
  <article>
   <h2>01</h2>
   <h3>1951</h3>
   <p>ALMEX ...</p>
   <picture></picture>
   <button>&lt;</button>
   <button>&gt;</button>
  </article>
 </section> <!-- Slider dentro de la sección principal. -->
</div>
<section class="noticias"></section>
<footer></footer>
```

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,338 @@
+        <h2>
+            NOTICIAS
+        </h2>
+        <article class="NoticiasPrincipal">
```
el nombre de las clases hay que usarlo en kebab-case

```html
<article class="noticias-principales">
```

semanticmx commented on template/index.html:
```diff
@@ -0,0 +1,338 @@
+        </article>
+    </section>
+    <footer>
+        <nav>
```
aquí como no hay links no va el nav, puede ir en un div contenedor.

semanticmx reviewed (approved):
Hay otros ajustes pero ya los vas a ocupar una vez que inicies con el CSS.

Solo revisa esos comentarios para que los vayas aplicando conforme aplicas CSS a esas secciones.
